test/acceptance/framework/k8s/debug.go

```diff
@@ -31,25 +31,51 @@ func WritePodsDebugInfoIfFailed(t *testing.T, kubectlOptions *k8s.KubectlOptions
 		testDebugDirectory := filepath.Join(debugDirectory, t.Name(), contextName)
 		require.NoError(t, os.MkdirAll(testDebugDirectory, 0755))
 
-		logger.Logf(t, "dumping logs and pod info for %s to %s", labelSelector, testDebugDirectory)
+		logger.Logf(t, "dumping logs, pod info, and envoy config for %s to %s", labelSelector, testDebugDirectory)
+
+		// Describe and get logs for any pods.
 		pods, err := client.CoreV1().Pods(kubectlOptions.Namespace).List(context.Background(), metav1.ListOptions{LabelSelector: labelSelector})
 		require.NoError(t, err)
 
 		for _, pod := range pods.Items {
 			// Get logs for each pod, passing the discard logger to make sure secrets aren't printed to test logs.
 			logs, err := RunKubectlAndGetOutputWithLoggerE(t, kubectlOptions, terratestLogger.Discard, "logs", "--all-containers=true", pod.Name)
-
-			// Write logs or err to file name <pod.Name>.log
-			logFilename := filepath.Join(testDebugDirectory, fmt.Sprintf("%s.log", pod.Name))
 			if err != nil {
 				logs = fmt.Sprintf("Error getting logs: %s: %s", err, logs)
 			}
+
+			// Write logs or err to file name <pod.Name>.log
+			logFilename := filepath.Join(testDebugDirectory, fmt.Sprintf("%s.log", pod.Name))
 			require.NoError(t, ioutil.WriteFile(logFilename, []byte(logs), 0600))
 
 			// Describe pod and write it to a file.
 			writeResourceInfoToFile(t, pod.Name, "pod", testDebugDirectory, kubectlOptions)
 		}
 
+		// Get envoy configuration from the mesh gateways, if there are any.
+		meshGatewayPods, err := client.CoreV1().Pods(kubectlOptions.Namespace).List(context.Background(), metav1.ListOptions{LabelSelector: "component=mesh-gateway"})
+		require.NoError(t, err)
+
+		for _, mpod := range meshGatewayPods.Items {
+			// Get configdump from mesh gateway, passing the discard logger since we only need these logs written to the file (below).
+			configDump, err := RunKubectlAndGetOutputWithLoggerE(t, kubectlOptions, terratestLogger.Discard, "exec", mpod.Name, "-c", "lifecycle-sidecar", "--", "curl", "-s", "localhost:19000/config_dump?format=json")
+			if err != nil {
+				configDump = fmt.Sprintf("Error getting config_dump: %s: %s", err, configDump)
+			}
+			// Get cluster config from mesh gateway, passing the discard logger since we only need these logs written to the file (below).
+			clusters, err := RunKubectlAndGetOutputWithLoggerE(t, kubectlOptions, terratestLogger.Discard, "exec", mpod.Name, "-c", "lifecycle-sidecar", "--", "curl", "-s", "localhost:19000/clusters?format=json")
+			if err != nil {
+				clusters = fmt.Sprintf("Error getting clusters: %s: %s", err, clusters)
+			}
+
+			// Write config/clusters or err to file name <envoy-[configdump/clusters]-pod.Name>.log
+			configDumpFilename := filepath.Join(testDebugDirectory, fmt.Sprintf("envoy-configdump-%s.json", mpod.Name))
+			clustersFilename := filepath.Join(testDebugDirectory, fmt.Sprintf("envoy-clusters-%s.json", mpod.Name))
```

Comment:
```suggestion
			// Write config/clusters or err to file name <pod.Name>-envoy-[configdump/clusters].json
			configDumpFilename := filepath.Join(testDebugDirectory, fmt.Sprintf("%s-envoy-configdump.json", mpod.Name))
			clustersFilename := filepath.Join(testDebugDirectory, fmt.Sprintf("%s-envoy-clusters.json", mpod.Name))
```

Since the pattern for the pod logs and describe is `<pod name>.log` and `<pod name>-pod.txt` I think the envoy pattern would be `<pod name>-envoy-clusters.json`?

Reply:
Yeah, that sounds good, thanks for catching that.